contents/blog/clickhouse-vs-bigquery.md

```diff
@@ -0,0 +1,168 @@
+
+![Granulated Storage.png](../images/blog/clickhouse-vs-bigquery/granulated-storage.png)
+
+Because data is sorted by primary key columns — and those columns are typically chosen based on what’s used to index data — ClickHouse needs to scan ***significantly*** less data for most queries. 
+
+### Materialized views
+
+Materialized views are psuedo-tables generated by other table data. Materialized views are available in a lot of databases, including industry-favorite, Postgres. However, most databases generate materialized views at a *specific* point in time. They aren’t updated automatically when new data is available; but in ClickHouse, they are. 
+
+ClickHouse doesn’t accomplish this by using a cron job. Instead, ClickHouse re-engineered how materialized views work. When data is ingested, typically in-bulk, it’s slowly added to a materialized view by merging with an intermediate data representation. 
+
+If you’ve ever implemented a MergeSort, ClickHouse’s underlying operation is similar. It is tuned for specific mathematical operations, like Mean or Count, and can append to a pre-existing materialized view incrementally.  
+
+### Specialized engines
+
+One of the magic features that powers materialized views is ClickHouse’s specialized engines. Specialized engines are engines that more efficiently utilize the disk for known mathematical patterns. For instance, when a materialized view is aided by ClickHouse’s `AggregatingMergeTree`, queries execute even faster. 
+
+This is one of the useful features we leverage extensively at PostHog – watch our **[video on them](https://youtu.be/6IwLWEx_mg4?t=804)** if you're curious.
+
+### Vectorized Query Execution
+
+Vectorized query execution allows ClickHouse to use SIMD (Single Instruction Multiple Data), which parallelize data aggregation of multiple memory values with a single instruction. This massively boosts computation. 
+
+While BigQuery doesn’t support vectorized query execution, Google does [offer it](https://cloud.google.com/bigquery/docs/bi-engine-query) through BI Engine, a separate GCP product that integrates with BigQuery. 
+
+## Why ClickHouse is often integrated with BigQuery
+
+ClickHouse and BigQuery are often used together, evidenced by ClickHouse’s [own blog post](https://clickhouse.com/blog/clickhouse-bigquery-migrating-data-for-realtime-queries) on the topic. In general, for GCP users, BigQuery offers an easy, consolidated place to analyze data given its easy integration with other GCP data products. 
+
+However, a tuned ClickHouse instance is faster. If analytical queries need to be returned to a customer without noticeable delays, ClickHouse is likely the better solution. It’s also probably cheaper, as BigQuery charges users per scan or requires pre-purchased slots, which quickly grows expensive – it’s exactly why [HIFI switched to ClickHouse](https://clickhouse.com/blog/hifis-migration-from-bigquery-to-clickhouse), for example.  
```

Comment:
Any insight on the pricing differences between the two?

Reply:
It is a bit of an apples to oranges comparison, but could mention it:
1. https://cloud.google.com/bigquery/pricing
2. https://clickhouse.com/pricing

Reply:
I think best to leave it. It's hard to compare + liable to change, dating the article.

Reply:
Fair point